src/main/java/com/wavefront/slug/chart/ChartSlugBuilderImpl.java

```diff
@@ -119,6 +127,18 @@ public ChartSlugBuilderImpl setBase(int base) {
     return this;
   }
 
+  @Override
+  public ChartSlugBuilder setChartSettings(String chartSettings) {
```

Comment:
why not just set this to ChartSettings?

Reply:
or compatible for both input

Reply:
I was concerned about exposing ChartSettings at first, as all other structured objects (Chart, ChartSlug) are considered internal / `package private`.
But maybe we should not have a fat builder and expose all of the objects in the first place?

Reply:
I think we want to make the user only create the slug through the builder interface.
If we expose ChartSettings and other classes, we won't be able to restrict this and the user is able to build a slug in a way we might not expect? Exposing chart settings only works, but is just not consistent with others.

Reply:
let's chat